fudge-ratio/src/main/webapp/code/time.js

```diff
@@ -1,52 +1,40 @@
 class Timer {
   seconds = 0;
-  minutes = 0;
-  hours = 0;
+  constructor(time = "00:00:00") {
+    const seconds = parseInt(time.substring(6, 8));
+    const minutes = 60 * parseInt(time.substring(3, 5));
+    const hours = 3600 * parseInt(time.substring(0, 2));
 
-  interval = null;
+    this.seconds = seconds + minutes + hours;
+  }
 
   // Concatenates a zero to the beginning of a number less than 10.
-  padd(num) {
+  padd(num = 0) {
     const padding = num < 10 ? "0" : "";
-    return padding + num;
-  }
-
-  constructor(time) {
-    if (time) {
-      this.hours = parseInt(time.substring(0, 2));
-      this.minutes = parseInt(time.substring(3, 5));
-      this.seconds = parseInt(time.substring(6, 8));
-    }
+    return padding + num.toString();
   }
 
-  allSeconds() {
-    return this.seconds + this.minutes * 60 + this.hours * 3600;
-  }
-
-  addOneSecond() {
-    this.seconds++;
-    if (this.seconds == 60) {
-      this.minutes++;
-      this.seconds = 0;
-    }
-    if (this.minutes == 60) {
-      this.hours = (this.hours + 1) % 24;
-      this.minutes = 0;
-    }
+  addSeconds(howMany = 0) {
```

Comment:
what does the variable howMany do here?

Reply:
we have a method to add seconds, howMany is just howMany seconds are we going to add